scitt_emulator/rkvst.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+from archivist.archivist import Archivist
+from typing import Optional
+from pathlib import Path
+import json
+import cbor2
+#from cose.messages import CoseMessage
+import base64
+from os import getenv
+from . import rkvst_mocks
+
+
+from scitt_emulator.scitt import SCITTServiceEmulator
+
+
+class RKVSTSCITTServiceEmulator(SCITTServiceEmulator):
+    tree_alg = "RKVST"
+
+    def __init__(
+        self, service_parameters_path: Path, storage_path: Optional[Path] = None
+    ):
+        super().__init__(service_parameters_path, storage_path)
+        if storage_path is not None:
+            self._service_private_key_path = (
+                self.storage_path / "service_private_key.pem"
+            )
+
+
+    def initialize_service(self):
+        #########################
+        # One time initial set-up
+        #########################
+
+        # TODO: is there anything permanent? 
+        # * Credentials? We refresh creds and conn each time.
+        # * Feeds? Feed is claim-specific
+
+        ###################
+        # Every time set-up
+        ###################
+
+        # Grab credentials from the enviroment
+        # TODO: we should support container storage and protected local file storage too
+        # TODO: we shold support unauthenticated connections for public read calls
```

Comment:
I had similar question, it seems like all api operations probably want some access control.

Reply:
Doing something sophisticated here needs resolution on the Registration Policies and Identifiers discussions I think. The signer of the Signed Statement is not necessarily the caller of the API (in fact, will never be in my book, otherwise you'll be reusing keys in a naughty way ;-) so there's some unpicking to do here.

But yes at the moment all API calls _do_ have authentication on them. My comment is that I think certain things could be relaxed in future when more of the API and surrounding concepts are fleshed out.